### ISSUE TITLE
RT-dev3065 crashes

### ISSUE THREAD

**Issue by marter001** [Beep6581/RawTherapee#6424]
I am working on windows 10 and download the RT dev-versions from https://keybase.pub/gaaned92/RTW64NightlyBuilds/
Version 3049 is working stable and only crashes very seldom when clicking the export button.
Version 3065 i cannot use at all, because it always crashes when clicking on the favorites folders in the explorer-tab. Have a look at this short video:

https://user-images.githubusercontent.com/78749513/152563497-b08eca0c-b01c-4df0-a33e-ffb8d063358b.mp4

On pixls i read, that a few other users had crashes of this version, but seems in a different manner. Thus i cannot give you a valid description to reproduce this crash.
Gaaned92 gave me an instruction to record the error messages, which i followed carefully:
https://discuss.pixls.us/t/thank-you-gaaned92-for-the-3058-windoze-builds/28222/20

https://discuss.pixls.us/t/problem-with-rt-5-8-3065/28872/16

Today i was able to do a test on two other windows machines and on both RT crashed too.
Since i dont know anything about builds: is there a chance, that these crashes only are based of this build?
Are there other builds for windows available, which i can make a test with?

**Comment by gaaned92** [comment]
@marter001

 I am unable to reproduce the crashes on my machine (W10, i7-6700K).

So the only way I see is to try to identify, if any, the offending commit.
For that, I consider a slow and laborious process which requires our collaboration: I am going to use  git-bisect method to find the offending commit by bisection.

1- I will produce a skylake build one at a time, that I will upload in https://keybase.pub/gaaned92/RTW64NightlyBuilds/marter/
2- I will notify you that the build is ready
3- You will download and test and indicate her if it is OK or buggy.
4-  we will iterate from 1 until finding the culprit (hoping it is due to a commit and not the MSYS2 updating!)

We should have a result in less than 5 iterations. 
The first step will be to find a good build. So I will first rebuild RawTherapee_dev_5.8-3049-gd053f0245_W64_skylake_210919 with the updated MSYS2.

Could you notify me if you agree to participate to the above process?
Can you indicate again the type of processor?

Thanks

**Comment by marter001** [comment]
@gaaned92 thanks a lot for your answer and wantig to take this effort. Of course I agree to your proposed process.

My machine:
Windows 10 21H1, I7-7700, 3.6 GHz
40 GB RAM
NVIDIA Geforce GTX 1060 3GB 
and (maybe important) 4K monitor BenQ, resolution 3840x2169

Thanks a lot in advance

**Comment by gaaned92** [comment]
After some verifications, I will try to start on Sunday.

**Comment by marter001** [comment]
great, but s long as default isn't revised we are not in a hurry :-)
Should i proceed the debug under GDB as before?
Please let me know if you need more information or some other assitance.

**Comment by rebio** [comment]
I tried to test this bug on both of my Windows machines.
**TLDR:** I was not able to reproduce it.

I tested the same build (Version: 5.8-3065-g9c5ce0d9b) from the same website.

**Machines:**
1) Windows 7 system: An old legacy system with a super old AMD CPU. I opened the Downloads folder in the favourites section. There was no crash no matter what I did (but I had to add the section first though).

2) Windows 10 system (Version 10.0.19043, Build 19043): A laptop with an AMD Ryzen 4000 series CPU. I again opened the Downloads folder in the favorites section (that I had the Dowloads folder again). There was again no crash no matter what I did. This download folder is quite the dump with many different file types (7z, doc, exe, jpg, mp4, ods, pdf, png, xcf, zip) and nothing caused any problems.


**Two short ideas:**
1) What if you removed all files from the folder? (Could the crash be caused by a file or file extension)
2) What if you removed the link to the Downloads folder from the favourites and readded it? (could the crash be caused by the link itself because it is eg. an invalid path or something like it)

**Comment by marter001** [comment]
@rebio thanks for testing. I tried to verify with your ideas, but not successfull.

The crash is not related to the download-folder it can be any folder:

- In my video you also can see the folder "urlaub", RT crashed when clicking this one after removing the download-folder.
- trying to remove all favorites already crashes in 3065, after restart they appear again
- The favorites are set up and overtaken from my stable version 3049, where no crash occurs, when clicking these folders.
- I removed all favorites with 3049 and restarted 3065
- clicking any folder in explorer tree caused a crash either (video also available)

I doubt, if my windows or hardware got a problem, because dev3049 is working fine.

**Comment by gaaned92** [comment]
@marter001 
Procedure
- Download the build in https://keybase.pub/gaaned92/RTW64NightlyBuilds/marter/ when I notify an upload
- test the build
- if it crashes, just post BAD here. if it doesn't crash just post GOOD here.

**Comment by gaaned92** [comment]
@marter001 
3065 uploaded


next time it will be 3049 (I will notify)

**Comment by Thanatomanic** [comment]
Before you proceed in this laborious effort, has @marter001 already tried to clear his cache? Just remove the entire cache folder from `%LOCALAPPDATA%/RawTherapee-5dev/` (exact path may differ, I'm on mobile, can't check).
Then restart RT and try to open folders.

**Comment by marter001** [comment]
@gaaned92 wow, you did it!

In a first test, there is no crash in this Version. I will continue and report.

@Thanatomanic yes i did this for former tests. With the today upload it was not necessary to delete any cache or folder.

BTW: I could find another unexpected behaviour on this favorite structure, but thats more like cosmetics: 
Click a folder in favorite, then click another folder in structure, the favorite folder keeps focussed and will not react on another click. First you need to click on a different favorite, then it will react as normal. cosmetics ;-)

Thanks a lot!

**Comment by gaaned92** [comment]
Thaks @marter001 
If there is no crash with 3065, there is no more to do for the moment on my side.
I am waiting for confirmation.

**Comment by marter001** [comment]
@gaaned92 ,thanks a lot to you.
I will give it a intensive test next weekend and in case of no crash, i will give you information and close this issue.

**Comment by Beep6581** [comment]
@marter001 can you give this build a try?
`RawTherapee_dev_WinVista_64_release.exe`
https://github.com/Beep6581/RawTherapee/releases/tag/nightly-github-actions

**Comment by marter001** [comment]
@Beep6581 thank for the link, i downloaded the file and will test it on weekend too.

**Comment by marter001** [comment]
@gaaned92, @Beep6581 thanks for your special builds.
I tested both of your builds and both are stable in explorer according to this issue.

But:
- In dev 3068 (@Beep6581) the lensfun doesn't work at all, its not possible to choose or define a camera or a lens.
- In both, and also in former dev (3049), i still get a seldom crash when clicking the export-button. 
I will more investigate in this and try to describe this in a separate issue.

**Comment by Thanatomanic** [comment]
The lensfun issue is known (but not officially reported). See here https://discuss.pixls.us/t/automated-windows-builds-commit-6413/29334/7

Please set the `DBDirectory` to what apostel338 mentions and see if that solves the issue.

**Comment by marter001** [comment]
@Thanatomanic thanks for the hint, in dev3068 this will work fine, but then in dev3065 the lensfun is gone...

**Comment by marter001** [comment]
no crashes anymore, so i think we can close this issue.
